### DISCUSSION TITLE
Dynamic Imports Not Working with fast-glob in Next.js on Vercel

### DISCUSSION THREAD

**Discussion by leadklasse** [vercel/community#5256] [Help]
### Summary

I'm experiencing an issue where dynamic imports using fast-glob in a Next.js application are working locally but not when deployed on Vercel. The application fails to dynamically load and process MDX files, resulting in empty data.

### Example

Unfortunately, I don't have a public link to a minimal reproduction. However, I can provide a snippet of the problematic code (see below).

### Steps to Reproduce

```
// mdx.ts

import { type ImageProps } from 'next/image'
import glob from 'fast-glob'

async function loadEntries<T extends { date: string }>(
  directory: string,
  metaName: string,
): Promise<Array<MDXEntry<T>>> {
  return (
    await Promise.all(
      (await glob('**/page.mdx', { cwd: `src/app/${directory}` })).map(
        async (filename) => {
          let metadata = (await import(`../app/${directory}/${filename}`))[
            metaName
          ] as T
          return {
            ...metadata,
            metadata,
            href: `/${directory}/${filename.replace(/\/page\.mdx$/, '')}`,
          }
        },
      ),
    )
  ).sort((a, b) => b.date.localeCompare(a.date))
}

type ImagePropsWithOptionalAlt = Omit<ImageProps, 'alt'> & { alt?: string }

export type MDXEntry<T> = T & { href: string; metadata: T }

export interface Article {
  date: string
  title: string
  description: string
  author: {
    name: string
    role: string
    image: ImagePropsWithOptionalAlt
  }
}

export interface CaseStudy {
  date: string
  client: string
  title: string
  description: string
  summary: Array<string>
  logo: ImageProps['src']
  image: ImagePropsWithOptionalAlt
  service: string
  testimonial: {
    author: {
      name: string
      role: string
    }
    content: string
  }
}

export function loadArticles() {
  return loadEntries<Article>('blog', 'article')
}

export function loadCaseStudies() {
  return loadEntries<CaseStudy>('work', 'caseStudy')
}

```

Environment Details:
Next.js version: 13.4.16
Vercel deployment
Relevant libraries: fast-glob, next/image

**Comment by KitsonBroadhurst** [comment]
Seeing something similar - any updates on this?

**Reply by leadklasse** [reply]
I found a solution to the problem I originally posted. The key was updating Next.js to version 14.0.4. This change resolved the dynamic import issue on Vercel. Hope this helps anyone facing the same issue!

**Reply by KitsonBroadhurst** [reply]
Thanks for the suggestion. I was on `14.1.0` already, but trying `14.0.4` didn't solve it either unfortunately.

**Reply by KitsonBroadhurst** [reply]
It seems there is still some open issue for this: https://github.com/vercel/next.js/issues/54804

**Reply by maichongju** [reply]
Do we have any update for this issue as I am having the same problem.

**Comment by maichongju** [comment]
I found the solution for it. Based on the [doc](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions). You will need to add addtional setting in your `next.config.js`

```
module.exports = {
  experimental: {
    outputFileTracingIncludes: {
      '/api/another': ['./necessary-folder/**/*'],
    },
  },
}
```
